pkg/plugin/datasource.go

```diff
@@ -133,6 +134,9 @@ func (d *Datasource) query(ctx context.Context, pCtx backend.PluginContext, quer
 		return nil, err
 	}
 
+	for key, value := range req.Headers {
```

Comment:
I'll admit I'm unfamiliar with this part of the backend, but I was looking around and I noticed that the function `GetHTTPHeaders()` also exists. I'm wondering if we are supposed to use that instead or what the different would be.

Reply:
You're right. The GetHTTPHeaders method filters unused headers ([source](https://github.com/grafana/grafana-plugin-sdk-go/blob/6ae7edaf09e3a007d3738e78b3492a4f6bb515b7/backend/http_headers.go#L66)). We can use that method for the better security.